src/optimizer.cpp

```diff
@@ -249,71 +250,99 @@ void Optimizer::propagateStateVelocitiesFromInitials(
 {
   motion_model_->predict(state);
 }
+
 void Optimizer::integrateStateVelocities(
-  xt::xtensor<float, 3> & trajectories,
-  const models::State & state) const
+  xt::xtensor<float, 2> & trajectories,
+  const xt::xtensor<float, 2> & state) const
 {
-  auto x = xt::view(trajectories, xt::all(), xt::all(), 0);
-  auto y = xt::view(trajectories, xt::all(), xt::all(), 1);
-  auto yaw = xt::view(trajectories, xt::all(), xt::all(), 2);
-
-  auto & w = state.wz;
   double initial_yaw = tf2::getYaw(state_.pose.pose.orientation);
 
-  yaw = utils::normalize_angles(xt::cumsum(w * settings_.model_dt, 1) + initial_yaw);
-  xt::xtensor<float, 2> yaw_offseted = yaw;
+  auto vx = xt::view(state, xt::all(), 0);
+  auto vy = xt::view(state, xt::all(), 2);
+  auto wz = xt::view(state, xt::all(), 1);
+
+  auto traj_x = xt::view(trajectories, xt::all(), 0);
+  auto traj_y = xt::view(trajectories, xt::all(), 1);
+  auto traj_yaws = xt::view(trajectories, xt::all(), 2);
 
-  xt::view(yaw_offseted, xt::all(), xt::range(1, _)) =
-    xt::view(yaw, xt::all(), xt::range(_, -1));
-  xt::view(yaw_offseted, xt::all(), 0) = initial_yaw;
+  traj_yaws = xt::cumsum(utils::normalize_angles(wz) * settings_.model_dt, 0) + initial_yaw;
```

Comment:
I believe this needs to be `utils::normalize_angles(wz * settings_.model_dt + initial_yaw)` so it can be normalized as angles after the time & offsets are applied

Reply:
Which actually the other implementation of this function does properly

Reply:
yeah that's a mistake